src/sql/workbench/services/accountManagement/browser/accountManagementService.ts

```diff
@@ -194,6 +197,17 @@ export class AccountManagementService implements IAccountManagementService {
 		return (<azdata.PromptFailedResult>result).canceled;
 	}
 
+	private isErrorResult(result: azdata.Account | azdata.PromptFailedResult): result is azdata.PromptFailedResult {
+		return (<azdata.PromptFailedResult>result).error;
+	}
+
+	private getMessage(result: azdata.Account | azdata.PromptFailedResult): string {
```

Comment:
You shouldn't need this with the type guard isErrorResult above - in the else if block it will already know it's a PromptFailedResult so the properties will be directly accessible.